--- /dev/null
+++ ../src/voku/SimplePhpParser/Model/PHPTrait.php
@@ -246,7 +246,7 @@
             $infoTmp['paramsPhpDocRaw'] = $paramsPhpDocRaw;
             $infoTmp['returnPhpDocRaw'] = $method->returnPhpDocRaw;
             $infoTmp['line'] = $method->line;
-            $infoTmp['file'] = $method->file;
+            $infoTmp['file'] = $method->file ?? $this->file;
             $infoTmp['error'] = \implode("\n", $method->parseError);
             foreach ($method->parameters as $parameter) {
                 $infoTmp['error'] .= ($infoTmp['error'] ? "\n" : '') . \implode("\n", $parameter->parseError);
